components/certificate-mgt/org.wso2.carbon.identity.certificate.management.core/src/main/java/org/wso2/carbon/identity/certificate/management/core/ApplicationCertificateManagementService.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.certificate.management.core;
+
+import org.wso2.carbon.identity.certificate.management.core.exception.CertificateMgtException;
+import org.wso2.carbon.identity.certificate.management.core.model.Certificate;
+
+/**
+ * This interface defines the Application Certificate Management Service.
+ * Application Certificate Management Service is the component that is responsible for managing certificates of
+ * the applications of a tenant.
+ * @deprecated Use {@link CertificateManagementService} instead.
```

Comment:
Let's mention why this is deprecated in the class comment as it is quite odd to add a new class that is deprecated

Reply:
Added with https://github.com/wso2/carbon-identity-framework/pull/6043/commits/1e9e867eddad241feca9e1d48b1a7cb053cf60fb